<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Ground Truth Localization -->
    <arg name="ground_truth" default="false"/>
    <group if="$(eval arg('ground_truth'))">
        <!-- Convert GPS data into Odometry data -->
        <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" output="screen">
            <rosparam command="load" file="$(find localization)/config/navsat.yaml"/>

            <remap from="/imu/data" to="/imu/ground_truth"/>
            <remap from="/gps/fix" to="/gps/ground_truth/coordinates"/>
            <!-- Should be from global ekf estimate for some reason -->
            <remap from="/odometry/filtered" to="/odometry/filtered/global"/>
        </node>

        <!-- EKF Local node, which fuses only contnuous data, i.e. no GPS -->
        <node pkg="robot_localization" type="ekf_localization_node" name="ekf_local" output="screen">
            <rosparam command="load" file="$(find localization)/config/ground_truth/ekf_local.yaml"/>
            <remap from="/odometry/filtered" to="/odometry/filtered/local"/>
        </node>

        <!-- EKF Global node, which incorporates discrete GPS data -->
        <node pkg="robot_localization" type="ekf_localization_node" name="ekf_global" output="screen">
            <rosparam command="load" file="$(find localization)/config/ground_truth/ekf_global.yaml"/>
            <remap from="/odometry/filtered" to="/odometry/filtered/global"/>
        </node>
    </group>

    <!-- Localization (with noise) -->
    <group unless="$(eval arg('ground_truth'))">
        <!-- Convert GPS data into Odometry data -->
        <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" output="screen">
            <rosparam command="load" file="$(find localization)/config/navsat.yaml"/>

            <remap from="/imu/data" to="/imu/data"/>
            <remap from="/gps/fix" to="/gps/coordinates"/>
            <!-- Should be from global ekf estimate for some reason -->
            <remap from="/odometry/filtered" to="/odometry/filtered/global"/>
        </node>

        <!-- EKF Local node, which fuses only contnuous data, i.e. no GPS -->
        <node pkg="robot_localization" type="ekf_localization_node" name="ekf_local" output="screen">
            <rosparam command="load" file="$(find localization)/config/noise/ekf_local.yaml"/>
            <remap from="/odometry/filtered" to="/odometry/filtered/local"/>
        </node>

        <!-- EKF Global node, which incorporates discrete GPS data -->
        <node pkg="robot_localization" type="ekf_localization_node" name="ekf_global" output="screen">
            <rosparam command="load" file="$(find localization)/config/noise/ekf_global.yaml"/>
            <remap from="/odometry/filtered" to="/odometry/filtered/global"/>
        </node>
    </group>
</launch>